cluster-autoscaler/cloudprovider/alicloud/alicloud_auto_scaling.go

```diff
@@ -166,14 +167,29 @@ func (m autoScalingWrapper) getScalingGroupByName(groupName string) (*ess.Scalin
 }
 
 func (m autoScalingWrapper) getScalingInstancesByGroup(asgId string) ([]ess.ScalingInstance, error) {
-	params := ess.CreateDescribeScalingInstancesRequest()
-	params.ScalingGroupId = asgId
-	resp, err := m.DescribeScalingInstances(params)
-	if err != nil {
-		klog.Errorf("failed to request scaling instances for %s,Because of %s", asgId, err.Error())
-		return nil, err
+	instances := make([]ess.ScalingInstance, 0)
+	pageNumber := 1
+
+	for {
+		params := ess.CreateDescribeScalingInstancesRequest()
+		params.ScalingGroupId = asgId
+		params.PageNumber = requests.NewInteger(pageNumber)
+		params.PageSize = requests.NewInteger(defaultRequestPageSize)
+		resp, err := m.DescribeScalingInstances(params)
```

Comment:
Should we add some unit tests around this? Since this is an unexported function, another option could be to test it by adding a test for `FindForInstance`. For starters, we can have a test related to the bug which fails for the old code.

Reply:
Thank you for your suggestion, I fully agree and will add unit tests as soon as possible.

Reply:
The unit tests have been added.